app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt

```diff
@@ -352,6 +351,16 @@ class EventDetailsFragment : Fragment() {
         menuActionBar = menu
     }
 
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        if (::eventShare.isInitialized) {
+            if (eventShare.externalEventUrl == null) {
+                menu.findItem(R.id.open_external_event_url).isVisible = false
+            }
+            setFavoriteIconFilled(eventShare.favorite)
```

Comment:
@nikit19 this statement is also to be executed only if `eventShare` has been initialised